### ISSUE TITLE
void checks - invalid lint or how to fix? 

### ISSUE THREAD

**Issue by zoechi** [dart-lang/linter#1082]
![image](https://user-images.githubusercontent.com/405837/42828523-ecedaeba-89e8-11e8-9cd2-2136598c2963.png)

`onPressed` is 

```
final VoidCallback onPressed;
...
typedef VoidCallback = void Function();
```

**Comment by a14n** [comment]
I'm not able to reproduce it on master with:

```dart
typedef VoidCallback = void Function();
issue_1082() {
  bool b;
  Future<T> pushNamed<T extends Object>(String routeName) => null;
  final VoidCallback onPressed = b ? () => pushNamed('') : null; // OK
}
```

Could you give this code a try in your project to see if a lint appears ?

**Comment by zoechi** [comment]
This code doesn't show the warning.

A reduced file where it shows the warning for me:

```dart
import 'package:flutter/material.dart';

class ImagePrompt extends StatelessWidget {
  const ImagePrompt({this.route = ''});

  final String route;

  @override
  Widget build(BuildContext context) => MaterialButton(
        onPressed:
            route != null ? () => Navigator.pushNamed(context, route) : null,
      );
}
```
```
[✓] Flutter (Channel master, v0.5.8-pre.72, on Mac OS X 10.13.6 17G65, locale en-AT)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
[✓] Android Studio (version 3.1)
[✓] IntelliJ IDEA Ultimate Edition (version 2018.1.6)
[✓] Connected devices (1 available)

• No issues found!
```

**Comment by a14n** [comment]
Thanks I can reproduce the issue with:

```dart
// issue 1082
typedef VoidCallback = void Function();
class C {
  C({VoidCallback f});
}
issue_1082() {
  bool b;
  Future<T> pushNamed<T extends Object>(String routeName) => null;
  new C(f: b ? () => pushNamed('') : null); // OK
}
```

**Comment by asashour** [comment]
I am unable to reproduce on `master`, the given case in https://github.com/dart-lang/linter/issues/1082#issuecomment-406212057 passes.

**Comment by pq** [comment]
Thanks for taking a look @asashour!
